homeassistant/components/hunterdouglas_powerview/cover.py

```diff
@@ -218,8 +272,20 @@ def _async_update_current_cover_position(self):
         """Update the current cover position from the data."""
         _LOGGER.debug("Raw data update: %s", self._shade.raw_data)
         position_data = self._shade.raw_data.get(ATTR_POSITION_DATA, {})
-        if ATTR_POSITION1 in position_data:
-            self._current_cover_position = int(position_data[ATTR_POSITION1])
+        # This could probably be cleaner but without access to every HD shade type I can't test it
+        if isinstance(self._shade, Silhouette):
```

Comment:
We should break the code up into a base class for existing shades and have a separate class for `Silhouette` shades

Reply:
youre right, I'll refactor this

Reply:
Further to that I think it would be better for a base shade class for tilt blinds that sits above Silohoutte

As Silohoutte is the Niche blind that differs from the other HD have to offer I think it should extend the base blind

most of your code would work as is - the other blinds simply have a max tilt that aligns to MAX_POSITION where the type 18/23 shades have a max tilt half that

if you look at the aiovapi api you will see they extend their base class for Silohoutte for this purpose only

Reply:
for context this is what i am suggesting (and used in the code i shared to the community)

class PowerViewShadeWithTilt(PowerViewShade):
    """Representation of a top down bottom up powerview shade."""

    def __init__(self, coordinator, device_info, room_name, shade, name):
        """Initialize the shade."""
        super().__init__(coordinator, device_info, room_name, shade, name)
        self._current_cover_tilt_position = MIN_POSITION

class PowerViewSilhouette(PowerViewShadeWithTilt):
    """Representation of a top down bottom up powerview shade."""

    def __init__(self, coordinator, device_info, room_name, shade, name):
        """Initialize the shade."""
        super().__init__(coordinator, device_info, room_name, shade, name)
        MAX_POSITION = 32767

Reply:
I agree, although each shade type then needs mapping to the appropriate class. I'm gonna leave that out of this PR so as not to hold it up. Once this code is in place this mapping can be added